### DISCUSSION TITLE
Renovate no longer running due to issues with packages

### DISCUSSION THREAD

**Discussion by Jamesking56** [renovatebot/renovate#29773] [Request Help] [answered]
### What would you like help with?

Configuration

### How are you running Renovate?

MEND Renovate App

### Please tell us more about your question or problem

Renovate seems to be having issues with a private Composer package (hosted on repman.io) and a private Docker image hosted on GitHub Packages.

![image](https://github.com/renovatebot/renovate/assets/253237/88e49c28-d834-4e2e-b130-6f5c810553c1)

### Logs (if relevant)

<details><summary>Logs</summary>

https://gist.github.com/Jamesking56/5e8b9536a2c7f17b5dd0e308bbcce7aa

</details>

**Comment by rarkins** [comment, marked as answer]
There's no sign of your config having `hostRules` for either of those hosts

**Reply by HonkingGoose** [reply]
To expand on the maintainers short answer a bit. They mean that you should tell Renovate:

- _where_ to fetch your private packages/dependencies
- how to _authenticate_ to the private registries

You can do this with the `hostRules` config option, maybe combined with `matchHost`.

Documentation:

- [Renovate docs, `hostRules` config option](https://docs.renovatebot.com/configuration-options/#hostrules)
- [Renovate docs, `matchHost` config option](https://docs.renovatebot.com/configuration-options/#matchhost).

**Reply by Jamesking56** [reply]
How come Renovate isn't automatically setting up credentials for ghcr.io?

> For GitHub Packages, Renovate will automatically provision hostRules for both ghcr.io (containers) ... This means that any private packages hosted on GitHub will be automatically authenticated if they are accessible using the same token by Renovate.
https://github.com/renovatebot/renovate/blob/main/docs/usage/getting-started/private-packages.md#github-packages

**Reply by rarkins** [reply]
These docs are out of date. We found that this functionality didn't work, including because GitHub app tokens don't work with github packages (a GitHub problem/limitation)

**Reply by HonkingGoose** [reply]
If the docs are out-of-date, we should update them. 😄 

I'm not sure _how_ to update:

- Drop the whole section about GitHub packages?
- Drop specific info from the section?
- Something else?

Right now, the docs say:

> ## Automatic hostRules credentials for platform-hosted registries
> 
> ### GitHub Packages
> 
> For GitHub Packages, Renovate will automatically provision hostRules for both `ghcr.io` (containers) and `*.pkg.github.com` (maven, npm, nuget, rubygems) using the GitHub platform token.
> This means that any private packages hosted on GitHub will be automatically authenticated if they are accessible using the same token by Renovate.
> 
> If you wish to _override_ this authentication by providing a different token, then your rule must be _at least as specific_ as the automatic rule that Renovate generates.
> For example:
> 
> ```json
> {
>   "hostRules": [
>     {
>       "matchHost": "npm.pkg.github.com",
>       "hostType": "npm",
>       "token": "some-personal-access-token"
>     }
>   ]
> }
> ```

**Reply by markscamilleri** [reply]
@rarkins sorry to hop on this discussion but I was going to post a question about this very same thing.

We were changing our configuration due to the removal of support for encrypted tokens, and based on this section in the documentation and the fact that Renovate requested new permissions (to read the private registry) lead us to believe this feature is working now.

However, it appears that's not the case. So does this still need to be removed? Is there a way we can use the app token to save us from managing a PAT?

(I'm happy to move this to a separate discussion if needs be)

**Reply by dlahn** [reply]
@markscamilleri Did you ever solve this?

**Reply by markscamilleri** [reply]
Hey @dlahn, unfortunately not at the moment. This feature was classed as experimental and therefore disabled in the GitHub hosted version of Renovate. I think if you have a self-hosted version, you can try this out by enabling the [`RENOVATE_X_GITHUB_HOST_RULES`](https://github.com/renovatebot/renovate/blob/main/lib/modules/platform/github/readme.md#package-registry-credentials) but as @rarkins said (do correct me if I'm wrong, please), the feature didn't work and that's why this became an "experimental" feature.


I'd suggest specifying your host rules manually and saving them in the repository. Most package managers have a config file where you can save and commit this information to the repository. Meanwhile, to allow Renovate to authenticate to a private registry - there's a separate discussion: https://github.com/renovatebot/renovate/discussions/33407#discussioncomment-11787771 

In my use case, since private packages are a minority of packages that I use, I have decided to update those manually and let Renovate update public packages.
